vllm/config.py

```diff
@@ -48,6 +47,8 @@
                         random_uuid, resolve_obj_by_qualname)
 
 if TYPE_CHECKING:
+    from dataclasses import dataclass
```

Comment:
why do we need to import the decorator from dataclasses?

I assume that `_typeshed` only fuly typed this decorator?

Reply:
mypy doesn't unserstand Pydantic dataclasses, so we use the builtin dataclass for type checking

Reply:
we need to include the extension from pydantic

https://docs.pydantic.dev/latest/integrations/mypy/

Reply:
Oh good suggestion, that does work, but there are ~20 other places in the codebase where our Pydantic models break mypy rules :(

I'm thinking of reverting the pydantic change in this PR so that I can get the last config standardisations in.

Reply:
Actually, I've managed to get this to work